src/gcs.rs

```diff
@@ -212,6 +212,67 @@ pub async fn get_bytes_with_client(
     }
 }
 
+/// Use simple_upload_object_with_client to directly upload objects without
+/// creating a resumable upload session. This is useful for small files in
+/// particular, as it reduces overhead compared with the resumable session.
+#[allow(dead_code)]
+// this isn't used by fs yet, but it probably will be. For now it is only used
+// by a test.
+pub async fn simple_upload_object_with_client(
```

Comment:
How would you feel about turning this into the logic in write_private_object for now? (open a cursor, append bytes, finalize). If we ultimately *do* find cases where we can know that we can do a single-shot upload / want it for internal use, we avoid making the "file handle".

Reply:
> internal use
That could be an interesting place to use single-shot uploads... intermediate objects like we might have for compose operations could be sped up a bit by using this kind of upload.

Anyway, I can do that, although perhaps it's better to just use the resumable upload function and get rid of this altogether?